environment/template.go

```diff
@@ -0,0 +1,314 @@
+package environment
+
```

Comment:
Not sure what's the need of doing all these re-factoring ??
I can see that part of `apply.go` file func are in this `template.go` file.

my point is if you want - rename_file / change_location .. but moving part of logic here to there is adding more efforts without any real benefit.

Reply:
which part are you talking about? I mean which part that would be related to the "apply" operation and not to processing templates or retrieving information from them. This file should contain only the logic that relates to templates and not to apply operation, or am I missing something?

Reply:
@MatousJobanek I have ran init_teant_test.go for both before and after this PR and compare response body for each request made during test, similar to what you have captured here ([before_link](https://gist.github.com/MatousJobanek/12d062712410349d1e980369a71f6181), [after_link](https://gist.github.com/MatousJobanek/3beefc66c8e98b147e68dd9f04188662)).

After capturing response_body on my local machine, I compared them and it looks good.  It has few differences wrt annotations and labels which you already mentioned.

So with this exercise, I see the changes are fine :+1: